### PR TITLE
WebKit version>2.38 needs unifdef and libavif during builds

### PR DESCRIPTION
**Pull request by woutervanh**
As already applied in webkitgtk

### CODE REVIEW

File: recipes-browser/wpewebkit/wpewebkit_2.40.1.bb [hunk 1]
```diff
@@ -9,7 +9,7 @@ SRC_URI = "https://wpewebkit.org/releases/${BPN}-${PV}.tar.xz;name=tarball \
 
 SRC_URI[tarball.sha256sum] = "c6b25e168b70f2121305ed078d0790e0aa4b0c73fce44e32ed42d4e5dd137ccb"
 
-DEPENDS += " libwpe"
+DEPENDS += " libwpe unifdef-native libavif"
```
Comment (manuelafm) [inline recipes-browser/wpewebkit/wpewebkit_2.40.1.bb:12]: Thanks for your submission, Wouter.

unifdev-native is already added later, at the end of the file, around line 38.

Comment (woutervanh) [reply recipes-browser/wpewebkit/wpewebkit_2.40.1.bb]: Ah, missed that one. I updated the PR

**Comment by philn** [comment]
imho avif should become a new PACKAGECONFIG.

**Comment by psaavedra** [comment]
> imho avif should become a new PACKAGECONFIG.

Yes, a packageconfig to set USE_AVIF ON/OFF and the corresponding depend with libavif.
 
https://github.com/WebKit/WebKit/blob/74b006a7fec7e2b21313fc6877e9200f7001d01e/Source/cmake/OptionsWPE.cmake#L47


Also, since this USE_AVIF is in WebKit at least since 2.34. I'd also suggest do it like this:  


``` sh
#in wpewebkit.inc
PACKAGECONFIG = "... avif"

PACKAGECONFIG[avif] = "-DUSE_AVIF=ON,-DUSE_AVIF=OFF,libavif`

```

**Comment by manuelafm** [comment]
Hi, I made changes to this in #468 but forgot to come back to this ticket... @woutervanh can you tell me if main is now working for you?  And sorry if I ran over this change, we were testing several things in parallel.

**Comment by woutervanh** [comment]
is ok

**Pull request closed without merging**